android/src/main/java/com/tailscale/ipn/NetworkChangeCallback.kt

```diff
@@ -9,34 +9,29 @@ import android.net.NetworkCapabilities
 import android.net.NetworkRequest
 import android.util.Log
 import libtailscale.Libtailscale
-import java.net.InetAddress
-import java.util.concurrent.locks.ReentrantLock
-import kotlin.concurrent.withLock
 
 object NetworkChangeCallback {
 
   private const val TAG = "NetworkChangeCallback"
 
-  private data class NetworkInfo(
-    var caps: NetworkCapabilities,
-    var linkProps: LinkProperties
-  )
+  private data class NetworkInfo(var caps: NetworkCapabilities, var linkProps: LinkProperties)
 
-  private val lock = ReentrantLock()
```

Comment:
I think we still need this; upon re-reading the docs, it looks like a call to `onCapabilitiesChanged` will not happen concurrently with `onLinkPropertiesChanged`, but only for the same interface; we could still see concurrent calls for two different interfaces. IMO, it's cheap enough to keep this lock and might save us from a subtle bug.